pkg/sql/colexec/external/external.go

```diff
@@ -61,6 +60,12 @@ func String(arg any, buf *bytes.Buffer) {
 
 func Prepare(proc *process.Process, arg any) error {
 	param := arg.(*Argument).Es
+	if proc.Lim.MaxMsgSize == 0 {
+		param.maxbatchSize = uint64(morpc.GetMessageSize())
+	} else {
+		param.maxbatchSize = proc.Lim.MaxMsgSize
+	}
+	param.maxbatchSize = uint64(float64(morpc.GetMessageSize()) * 0.8)
```

Comment:
this is simply wrong.   maxbatchSize is overwritten so the if else is noop.

maxBatchSize please.